torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -97,23 +108,44 @@ template<typename... Args> inline variable_list flatten_tensor_args(Args&&... ar
 }
 
 // See NOTE [ Autograd View Variables ] for details.
-inline Tensor as_view(const Tensor & base, Tensor tensor, bool is_differentiable = true) {
+inline Tensor as_differentiable_view(const Tensor & base, Tensor tensor, bool allow_rebase_history) {
```

Comment:
Duplication here is not great

Reply:
I wasn't sure how to handle the fact that differentiable views take the `allow_rebase_history` argument but not the non-differentiable ones.
You would prefer that I merge both back and add the `allow_rebase_history` to both (asserting that it is `true` for non differentiable views)?

Reply:
Yes, I think that's my preference. If you want to deal with boolean blindness, make a little enum and pass that in as the input rather than just a boolean true or false.